### PR TITLE
Implemented WritableBitmap and bitmap creation by copying from in-memory data

### PR DESCRIPTION
**Pull request by kekekeks**

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
Any chance we can also add the capability to create a bitmap from a locked framebuffer? That would also be really nice for AvaloniaAV. (In any case this is really nice too, but has an additional CPU memory copy step).

**Comment by kekekeks** [comment]
@jkoritzinsky 
The problem with creating bitmap directly from user-provided memory is that we need to somehow ensure that it's alive the whole time. We'll be getting hard-to-debug native crashes otherwise.

In case of framebuffer it's used as an output surface and `DrawingContext` has predictable lifetime, so in that case it's somewhat (but not completely) safe.

**Comment by jkoritzinsky** [comment]
I can understand that. It also looks like you added the ability to construct a Bitmap from raw pixel data (which basically does what I wanted anyway) which is awesome. Thanks!

File: src/Windows/Avalonia.Direct2D1/Media/Imaging/WicBitmapImpl.cs [hunk 1]
```diff
@@ -53,17 +56,38 @@ public WicBitmapImpl(ImagingFactory factory, Stream stream)
         /// <param name="factory">The WIC imaging factory to use.</param>
         /// <param name="width">The width of the bitmap.</param>
         /// <param name="height">The height of the bitmap.</param>
-        public WicBitmapImpl(ImagingFactory factory, int width, int height)
+        /// <param name="pixelFormat">Pixel format</param>
+        public WicBitmapImpl(ImagingFactory factory, int width, int height, APixelFormat? pixelFormat = null)
```
Comment (grokys) [inline src/Windows/Avalonia.Direct2D1/Media/Imaging/WicBitmapImpl.cs]: This is an enum so why not just do `APixelFormat pixelFormat = APixelFormat.Bgra8888`?

**Review by grokys — APPROVED** [review]
Looks good other than one really minor question! I've not had chance to test it, @jkoritzinsky I assume you've tested it from your comments?

**Comment by jkoritzinsky** [comment]
I haven't had a chance to test but the API shape seems to be what Im looking for. I've been extremely busy recently so I haven't had much time for AvaloniaAV over the last few weeks or so.

File: src/Windows/Avalonia.Direct2D1/Media/Imaging/WicBitmapImpl.cs (hunk 1 above)
Comment (kekekeks) [inline src/Windows/Avalonia.Direct2D1/Media/Imaging/WicBitmapImpl.cs]: That's for consistency with [WritableBitmap.ctor](https://github.com/AvaloniaUI/Avalonia/pull/903/files#diff-b5070a10b906143832bc216ca08b83bdR15) and [CreateWritableBitmap](https://github.com/AvaloniaUI/Avalonia/pull/903/files#diff-bae6bd8ba7fde4a36d352d35193f2648R66). So we could just pass `null` through the call chain and don't worry about default pixel format outside of `WicBitmapImpl`.

**Pull request merged**